agent/consul/rpc.go

```diff
@@ -542,12 +540,7 @@ func canRetry(args interface{}, err error) bool {
 
 	// Reads are safe to retry for stream errors, such as if a server was
 	// being shut down.
-	info, ok := args.(structs.RPCInfo)
-	if ok && info.IsRead() && lib.IsErrEOF(err) {
-		return true
-	}
-
-	return false
+	return info != nil && info.IsRead() && lib.IsErrEOF(err)
```

Comment:
I'm getting a funny feeling that this may have the nil-not-nil behavior after the refactor. I could totally be wrong though.

Reply:
Nah I think it's fine: https://play.golang.org/p/3oTfQl1EQ4G

Reply:
Nice test! That's a good thing to confirm.